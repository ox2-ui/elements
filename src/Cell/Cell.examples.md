### Cell

    <Cell customStyle={{ padding: '30px' }}>
      <div>Child</div>
    </Cell>
