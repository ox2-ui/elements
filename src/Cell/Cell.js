import React, { PropTypes } from 'react';

const styles = {
  base: {
    display: 'table-cell',
  },
};

/**
 * Cell Component
 */
const Cell = ({ className, style, children, customStyle }) => (
  <div
    className={className}
    style={{ ...styles.base, ...style, ...customStyle }}
  >
    {children}
  </div>
);


Cell.propTypes = {
  /**
   * Cell content.
   */
  children: PropTypes.node.isRequired,
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * Cell specific style properties.
   */
  customStyle: PropTypes.object.isRequired,
  /**
   * Override the inline-styles of the text element.
   */
  style: PropTypes.object,
};

export default Cell;
