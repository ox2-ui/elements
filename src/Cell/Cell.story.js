import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Cell from './Cell';

storiesOf('Cell', module)
  .add('default', () => (
    <Cell customStyle={{ padding: '30px' }}>
      <div>Child</div>
    </Cell>
  ));
