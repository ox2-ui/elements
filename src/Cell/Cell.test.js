/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Cell from './Cell';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <Cell customStyle={{ padding: '30px' }}>
      <div>Child</div>
    </Cell>
  );

  expect(wrapper).toMatchSnapshot();
});
