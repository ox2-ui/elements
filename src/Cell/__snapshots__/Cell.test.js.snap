// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Cell
  customStyle={
    Object {
      "padding": "30px",
    }
  }
>
  <div
    style={
      Object {
        "display": "table-cell",
        "padding": "30px",
      }
    }
  >
    <div>
      Child
    </div>
  </div>
</Cell>
`;
